qutebrowser/browser/hints.py

```diff
@@ -56,6 +56,7 @@ def on_mode_entered(mode, win_id):
                       maybe=True)
 
 
+_reusable_hint_labels = []
```

Comment:
The "formal" name for this kind of pattern is an 'object pool', so maybe this could be named `_hint_label_pool` or similar?

Maybe this could be a class variable as well, instead of in the module (that would allow moving `create_hint_label` into `HintLabel` as well)